File: python-examples/captcha-solving-auth-example/auth-sessions/create.py
from typing import TypedDict

from intuned_runtime.captcha import wait_for_captcha_solve
from playwright.async_api import Page


class Params(TypedDict):
    email: str
    password: str


async def create(page: Page, params: Params | None = None, **_kwargs):
    if params is None:
        raise ValueError("Params with email and password are required")
    
    await page.goto("https://www.scrapingcourse.com/login/cf-turnstile")
    await page.locator("#email").fill(params["email"])
    await page.locator("#password").fill(params["password"])
    await wait_for_captcha_solve(page, timeout_s=30, settle_period=10)
    await page.locator("#submit-button").click()
    # Verify successful login by checking if Logout button is visible
    # If the Logout button is not visible, wait_for will raise an exception
    await page.get_by_text("Logout").wait_for(state="visible", timeout=10_000)

File: typescript-examples/captcha-in-login/auth-sessions/create.ts
import { Page, BrowserContext } from "playwright";
import { goToUrl } from "@intuned/browser";
import { waitForCaptchaSolve } from "@intuned/runtime";

export interface CreateAuthSessionParams {
  email: string;
  password: string;
}

export default async function create(
  params: CreateAuthSessionParams,
  page: Page,
  context: BrowserContext
): Promise<void> {
  await goToUrl({
    page,
    url: "https://www.scrapingcourse.com/login/cf-turnstile",
  });

  await page.locator("#email").fill(params.email);
  await page.locator("#password").fill(params.password);

  await waitForCaptchaSolve(page, {
    timeoutInMs: 30_000,
    settleDurationMs: 10_000,
  });

  await page.locator("#submit-button").click();
  // Verify successful login by checking if Logout button is visible
  // If the Logout button is not visible, waitFor will raise an exception
  await page.getByText("Logout").waitFor({ state: "visible", timeout: 10_000 });
}
